rust/core-lib/src/editor.rs

```diff
@@ -663,13 +670,17 @@ impl<W: Write + Send + 'static> Editor<W> {
         // TODO: we should probably bubble up this error now. in the meantime always set path,
         // because the caller has updated the open_files list
         self.path = Some(path.as_ref().to_owned());
+        self.save_rev_id = self.last_rev_id;
+        self.view.set_dirty();
+        self.render();
     }
 
     fn do_scroll(&mut self, first: i64, last: i64) {
         let first = max(first, 0) as usize;
         let last = last as usize;
+        let unsaved_changes = self.has_unsaved_changes();
```

Comment:
Hmm, now I'm a little more uncomfortable and would prefer this to be stashed. do_scroll() can happen with very high frequency. Plus, stashing the state might decrease the amount of threading of has_unsaved_changes as a parameter into method calls. (thinking about it, that's more true if the boolean lives in View than if it lives in Editor; conceptually you can make a case for either, so i'd go with whichever is less code)